extension/src/Error.ts

```diff
@@ -0,0 +1,11 @@
+export class InvalidXmlError extends Error {
+  constructor(
+    msg: string,
+    public path: string,
+    public index: number,
+    public length: number
+  ) {
+    super(msg);
+    Object.setPrototypeOf(this, InvalidXmlError.prototype);
```

Comment:
Apparently it's not necessary to use `setPrototypeOf` anymore (since ES6) so it's been removed. I think it makes the error be an instance of `InvalidXmlError` instead of `Error`. Only guessing tho :)